plugins/variational_quantum_classifier/tasks.py

```diff
@@ -0,0 +1,327 @@
+
+import os
+from tempfile import SpooledTemporaryFile
+
+from typing import Optional, List
+
+from celery.utils.log import get_task_logger
+
+from . import VQC
+
+from .schemas import (
+    InputParameters,
+    InputParametersSchema,
+)
+from qhana_plugin_runner.celery import CELERY
+from qhana_plugin_runner.db.models.tasks import ProcessingTask
+from qhana_plugin_runner.plugin_utils.entity_marshalling import (
+    save_entities,
+    load_entities,
+    ensure_dict,
+)
+from qhana_plugin_runner.requests import open_url
+from qhana_plugin_runner.storage import STORE
+
+import numpy as np
+
+from qiskit.utils import QuantumInstance
+from .backend.vqc import qiskitVQC
+
+from sklearn.metrics import accuracy_score
+
+from .backend.visualization import plot_data, plot_confusion_matrix
+
+TASK_LOGGER = get_task_logger(__name__)
+
+
+def get_point(ent):
+    dimension_keys = list(ent.keys())
+    dimension_keys.remove("ID")
+    dimension_keys.remove("href")
+
+    dimension_keys.sort()
+    point = np.empty(
+        (
+            len(
+                dimension_keys,
+            )
+        )
+    )
+    for idx, d in enumerate(dimension_keys):
+        point[idx] = ent[d]
+    return point
+
+
+def get_entity_generator(entity_points_url: str):
+    """
+    Return a generator for the entity points, given an url to them.
+    :param entity_points_url: url to the entity points
+    """
+    file_ = open_url(entity_points_url)
+    file_.encoding = "utf-8"
+    file_type = file_.headers["Content-Type"]
+    entities_generator = load_entities(file_, mimetype=file_type)
+    entities_generator = ensure_dict(entities_generator)
+    for ent in entities_generator:
+        yield {"ID": ent["ID"], "href": ent.get("href", ""), "point": get_point(ent)}
+
+
+def get_indices_and_point_arr(entity_points_url: str) -> (dict, List[List[float]]):
+    entity_points = list(get_entity_generator(entity_points_url))
+    id_to_idx = {}
+
+    idx = 0
+
+    for ent in entity_points:
+        if ent["ID"] in id_to_idx:
+            raise ValueError("Duplicate ID: ", ent["ID"])
+
+        id_to_idx[ent["ID"]] = idx
+        idx += 1
+
+    points_cnt = len(id_to_idx)
+    dimensions = len(entity_points[0]["point"])
+    points_arr = np.zeros((points_cnt, dimensions))
+
+    for ent in entity_points:
+        idx = id_to_idx[ent["ID"]]
+        points_arr[idx] = ent["point"]
+
+    return id_to_idx, points_arr
+
+
+def get_label_generator(entity_labels_url: str):
+    """
+    Return a generator for the entity labels, given an url to them.
+    :param entity_labels_url: url to the entity labels
+    """
+    file_ = open_url(entity_labels_url)
+    file_.encoding = "utf-8"
+    file_type = file_.headers["Content-Type"]
+    entities_generator = load_entities(file_, mimetype=file_type)
+    entities_generator = ensure_dict(entities_generator)
+    for ent in entities_generator:
+        yield {"ID": ent["ID"], "href": ent.get("href", ""), "label": ent["label"]}
```

Comment:
The expected format of the labels should be documented somewhere, i.e. that the label needs to be in the attribute "label" and that it needs to be an integer.

Reply:
I changed the code and the label data type, such that labels can now be strings aswell, e.g. "dog" or "cat". I documented the data type in the docs entities.rst. I would like the labels to not be restricted to strings, but also to include integers or other data types, but this seems to be very "unclean". What do you think?

Reply:
I think it would be fine to allow arbitrary data types as labels. With the label_to_int and int_to_label dicts you already have what you need for this. You probably only need to remove the string convertion `label = str(ent["label"])`.